Compilebox integration

leetim opened the pull request:
Готово, хотя автоматическое тестирование большой проблема.

Resolves #41

leetim committed:
Add menu item in interface

leetim committed:
Compile box integration done!

Added:
** middlware for compileBox
** dependency on react-redux-fetch

Changed:
** reducer chat (added handing for command POST_FULFIL)

leetim committed:
Merge branch 'master' of https://github.com/aberezkin/codecolla into 41-compilebox-integration

aberezkin commented on src/actions/index.js:
```diff
@@ -112,10 +113,27 @@ export function addMessage(message) {
 }
 
+export const COMPILE_CODE = 'COMPILE CODE';
+export function sendCode(langid, text) {
```
Мне кажется эту функцию можно протестировать.

aberezkin commented on src/middleware/index.js:
```diff
@@ -3,8 +3,10 @@ import multi from 'redux-multi';
+import {middleware as fetchMiddleware} from 'react-redux-fetch';
 
-const middlewares = [peers, chat, text, multi];
+const middlewares = [peers, chat, text, multi, compileCode, fetchMiddleware];
```
Тут лучше, чтобы multi был в именно в конце. Этот `middleware` обрабатывает массивы, так что если какой-то из `middleware` после него диспатчит массив, он не обработается.

И наверное стоит написать коммент об этом.

aberezkin commented on src/middleware/compile-code.js:
```diff
@@ -0,0 +1,54 @@
+                next(toggleChatAction);
+            }
+
+            const addMessageAction = addMessage(createMessage('Code is sending'));
```
> Code is sent

aberezkin commented on src/middleware/compile-code.js:
```diff
@@ -0,0 +1,54 @@
+    date: new Date(),
+});
+const getText = state => state.text.toArray().map(i => i.toObject().text).join('\n');
+const getLanguage = (actionType) => {
```
Наверное было бы удобно сделать `Map`, в котором бы были доступны `id` языков по их названию. У нас в state есть язык, который сейчас выбран в качестве основного в редакторе и таким образом можно было бы автоматически определять, какой id при компиляции отправлять в `CompileBox`.

aberezkin commented on src/reducers/chat.js:
```diff
@@ -8,6 +14,14 @@ const messages = (state = [], action) => {
             ];
+
+        case POST_FULFIL:
+            console.log(action.value.errors);
```
Уберай за собой после дебага :)

leetim committed:
Merge branch 'master' of https://github.com/aberezkin/codecolla into 41-compilebox-integration

leetim committed:
Added some changes for compilingCode

*Now compile menuItem select language_id by syntax highlight
*Added action-test for sendCode

leetim commented:
Поправил! Смотри)

aberezkin commented:
Код хороший. На паре сегодня посмотрим как это работает :)

kofes committed:
fix problem with Menu-style(firefox) & fix problem with key-warnings(compile-box)

leetim committed:
Change names by compileBox messages.

leetim committed:
Some little change.
